web-common/src/runtime-client/watch-request-client.ts

```diff
@@ -33,17 +37,38 @@ export class WatchRequestClient<Res extends WatchResponse> {
   ) {}
 
   public start(): Unsubscriber {
-    const unsubscribe = runtime.subscribe((runtimeState) => {
-      if (
-        !runtimeState ||
-        (runtimeState.instanceId === this.prevInstanceId &&
-          runtimeState.host === this.prevHost)
-      ) {
+    const store = derived([runtime, pageInFocus], (state) => state);
+    const unsubscribe = store.subscribe(([runtimeState, pageInFocus]) => {
+      const runtimeUnchanged =
+        runtimeState.instanceId === this.prevInstanceId &&
+        runtimeState.host === this.prevHost;
+      // const focusUnchanged = pageInFocus === this.prevFocus;
+
+      if (!runtimeState || runtimeUnchanged || !pageInFocus) {
+        if (!pageInFocus) {
+          this.prevInstanceId = this.prevHost = undefined;
+          this.prevFocus = false;
+          // cancel the watcher if page is not in focus
+          this.outOfFocusThrottler.throttle(() => {
+            this.controller?.abort();
+          });
+        }
         return;
       }
+      if (!this.prevFocus) {
+        // Call onReconnect on page focus to make sure we didnt miss anything
+        this.onReconnect();
+      }
       this.prevInstanceId = runtimeState.instanceId;
       this.prevHost = runtimeState.host;
+      this.prevFocus = true;
 
+      if (this.outOfFocusThrottler.throttling()) {
+        // Cancel any callbacks for out of focus
+        this.outOfFocusThrottler.cancel();
+        // The client is already running. Do not cancel
```

Comment:
I don't understand this comment given the `cancel()` method above

Reply:
The comment is for the watcher client.